### DISCUSSION TITLE
rewrites - support for http methods other than get

### DISCUSSION THREAD

**Discussion by mvording** [vercel/next.js#21155] [Ideas]
**Describe the feature you'd like to request**

It would be useful to be able to redirect all http methods using the Redirects feature (https://nextjs.org/docs/api-reference/next.config.js/redirects).

Usage:
/api/:path*   requests sent to another server for all REST HTTP methods.    

For server deployments this redirect is done by the ingress/load balancer, but for local development ideally next.js can redirect all requests to another application endpoint.    This works fine for GET requests, but not for other HTTP methods.

**Describe the solution you'd like**
```
  async rewrites() {
    return [
      {
        source: '/api/:path*',
        destination: 'http://localhost:8080/api/:path*',
        methods: [ '*']  or [ 'GET','POST','PUT','DELETE'] 
      },
    ]
  }
```
**Describe alternatives you've considered**

.

**Comment by quebecCu** [comment]
up
